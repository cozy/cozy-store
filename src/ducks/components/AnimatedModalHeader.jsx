import React from 'react'
import { AnimatedContentHeader } from 'cozy-ui/react/Modal'
import { Placeholder } from 'ducks/components/AppsLoading'
import Icon from 'cozy-ui/react/Icon'

import CozyCircle from 'assets/icons/cozy-blue-circle.svg'
import defaultAppIcon from 'assets/icons/icon-cube.svg'

// We have to use it as a classical function since the
// Modal from cozy-ui needs to explicitely find the
// AnimatedContentHeader in the direct children
// Using a component will hide the AnimatedContentHeader
// in the component children and the Modal from cozy-ui
// won't be able to find it
export const AnimatedModalHeader = ({ app }) => (
  <AnimatedContentHeader
    className="sto-animated-header"
    activeClassName="sto-animated-header--active"
  >
    {app.iconToLoad ? (
      <div className="sto-animated-header-icon">
        <Placeholder width="3.5rem" height="3.5rem" />
      </div>
    ) : app.icon ? (
      <img
        src={app.icon}
        alt={`${app.slug}-icon`}
        width="56"
        height="56"
        className="sto-animated-header-icon"
      />
    ) : (
      <Icon
        className="sto-animated-header-icon blurry"
        width="48"
        height="64"
        icon={defaultAppIcon}
        color="#95999D"
      />
    )}
    <Icon
      icon="exchange"
      color="#95999D"
      className="sto-animated-header-exchange"
    />
    <Icon
      icon={CozyCircle}
      className="sto-animated-header-icon"
      width="56"
      height="56"
    />
  </AnimatedContentHeader>
)

export default AnimatedModalHeader
