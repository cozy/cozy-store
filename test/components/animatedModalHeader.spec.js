'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import AnimatedModalHeader from 'ducks/components/AnimatedModalHeader'

Enzyme.configure({ adapter: new Adapter() })

describe('AnimatedModalHeader component', () => {
  it('should be rendered correctly if icon', () => {
    const app = { slug: 'collect', icon: 'icon.svg' }
    const component = shallow(<AnimatedModalHeader app={app} />).getElement()
    expect(component).toMatchSnapshot()
  })
  it('should handle correctly if icon is loading', () => {
    const app = { slug: 'collect', iconToLoad: true }
    const component = shallow(<AnimatedModalHeader app={app} />).getElement()
    expect(component).toMatchSnapshot()
  })
  it('should return default icon if no app icon provided', () => {
    const app = { slug: 'collect' }
    const component = shallow(<AnimatedModalHeader app={app} />).getElement()
    expect(component).toMatchSnapshot()
  })
})
