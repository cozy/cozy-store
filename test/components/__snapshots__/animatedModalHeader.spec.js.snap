// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedModalHeader component should be rendered correctly if icon 1`] = `
<AnimatedContentHeader
  activeClassName="sto-animated-header--active"
  className="sto-animated-header"
>
  <img
    alt="collect-icon"
    className="sto-animated-header-icon"
    src="icon.svg"
  />
  <Icon
    className="sto-animated-header-exchange"
    color="#95999D"
    icon="exchange"
    spin={false}
  />
  <Icon
    className="sto-animated-header-icon"
    height="56"
    icon="test-file-stub"
    spin={false}
    width="56"
  />
</AnimatedContentHeader>
`;

exports[`AnimatedModalHeader component should handle correctly if icon is loading 1`] = `
<AnimatedContentHeader
  activeClassName="sto-animated-header--active"
  className="sto-animated-header"
>
  <div
    className="sto-animated-header-icon"
  >
    <Placeholder
      height="3.5rem"
      width="3.5rem"
    />
  </div>
  <Icon
    className="sto-animated-header-exchange"
    color="#95999D"
    icon="exchange"
    spin={false}
  />
  <Icon
    className="sto-animated-header-icon"
    height="56"
    icon="test-file-stub"
    spin={false}
    width="56"
  />
</AnimatedContentHeader>
`;

exports[`AnimatedModalHeader component should return default icon if no app icon provided 1`] = `
<AnimatedContentHeader
  activeClassName="sto-animated-header--active"
  className="sto-animated-header"
>
  <Icon
    className="sto-animated-header-icon blurry"
    color="#95999D"
    height="64"
    icon="test-file-stub"
    spin={false}
    width="48"
  />
  <Icon
    className="sto-animated-header-exchange"
    color="#95999D"
    icon="exchange"
    spin={false}
  />
  <Icon
    className="sto-animated-header-icon"
    height="56"
    icon="test-file-stub"
    spin={false}
    width="56"
  />
</AnimatedContentHeader>
`;
